### ISSUE TITLE
Feature Request:  128-bit floats

### ISSUE THREAD

**Issue by johnpjust** [tensorflow/tensorflow#19421]
I have posted a question and answer on [SO](https://stackoverflow.com/questions/50438071/cost-function-convergence-in-tensorflow-using-softmax-cross-entropy-with-logits/50438072#50438072) showing why this is needed.  Admittedly it is probably not a common case, but I do research and the problems I deal with usually require creative solutions and so I'm generally pushing the boundaries of packages like Tensorflow (which is a wonderful package!).  I ran into this with soft-target classification.  

> with soft targets, especially ones that aren't close to 1 or zero, cross entropy loss doesn't change significantly as the algorithm improves. Let's say the targets are [0.39019628, 0.44301641, 0.16678731]. Well, using the formula for cross entropy

`cross_entropy = tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(y), reduction_indices=[1]))`

> but then using the targets "y_" in place of the predicted probabilities "y" we arrive at the true entropy value of 1.0266190072458234. If you're predictions are just slightly off of target....lets say they are [0.39511779, 0.44509024, 0.15979198], then the cross entropy is 1.026805558049737.

Basically even with 64-bit floats, the loss function shows evidence that it would continue to train if further significant digits were available.  This is just to support the cost function during the training process since the final trained values would not need such precision, but to get to an optimal convergence I need it.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 96 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 111 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 126 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 141 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 156 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 171 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 186 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tilakrayal** [comment]
@johnpjust ,
Is this still an issue?
Could you update TensorFlow to the latest stable version v.2.9 and let us know if you are facing the same error. Thank you!

**Comment by johnpjust** [comment]
This wasn't a Tensorflow issue necessarily, but I can close since it's a very rare need.

**Comment by johnpjust** [comment]
Very rare case.

**Comment by GitKalo** [comment]
I know this is closed, but just mentioning that I would like to see this in TF as well. For research purposes, I need to examine the effect of very small changes to the weights, and it would be great to go beyond the 64-bit double precision.
